test/Microsoft.Azure.Devices.Edge.Test/Context.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test
+{
+    using System;
+    using System.IO;
+    using System.Net;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Configuration;
+
+    public class Context
+    {
+        static readonly Lazy<Context> Default = new Lazy<Context>(
+            () =>
+            {
+                IConfiguration context = new ConfigurationBuilder()
+                    .SetBasePath(Directory.GetCurrentDirectory())
+                    .AddJsonFile("context.json")
+                    .AddEnvironmentVariables("E2E_")
+                    .Build();
+
+                string GetString(string name) => context.GetValue<string>(name);
+
+                string GetStringOr(string name, string alternate)
+                {
+                    string value = GetString(name);
+                    return string.IsNullOrEmpty(value) ? alternate : value;
+                }
+
+                Option<(string, string, string)> AllOrNothing(string a, string b, string c) =>
+                    string.IsNullOrEmpty(a) || string.IsNullOrEmpty(b) || string.IsNullOrEmpty(c)
+                        ? Option.None<(string, string, string)>()
+                        : Option.Some((a, b, c));
+
+                return new Context
```

Comment:
Nit: If you do this in the ctor, you won't need the private setters.

Reply:
I had a ctor originally, but when it got to 17 parameters I got rid of it. 😃

Reply:
No, I meant doing all of this in the ctor. But this is fine.